core/common/src/main/java/alluxio/network/protocol/databuffer/DataBuffer.java

```diff
@@ -40,6 +40,22 @@
   ByteBuffer getReadOnlyByteBuffer();
 
   /**
+   * Transfers this buffer's data to the specified destination starting at
+   * the current {@code readerIndex} and increases the {@code readerIndex}
+   * by the number of the transferred bytes (= {@code length}).
+   *
+   * @param dst the destination
+   * @param dstIndex the first index of the destination
+   * @param length the number of bytes to transfer
+   */
+  void readBytes(byte[] dst, int dstIndex, int length);
+
+  /**
+   * @return the number of readable bytes remaining
+   */
+  int readableBytes();
```

Comment:
https://docs.oracle.com/javase/specs/jls/se8/html/jls-10.html#jls-10.4

Arrays must be indexed by int values; short, byte, or char values may also be used as index values because they are subjected to unary numeric promotion (§5.6.1) and become int values.